### PR TITLE
chore(stack): validate path and root_directory against allowed characters

### PR DESCRIPTION
**Pull request by bvtujo**
This PR adds limits to the length of input paths corresponding to the limits of docker containers and EFS volumes. It also ensures that paths do not include special characters beyond a small allowlist of common characters and the `/` separator. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/deploy/cloudformation/stack/transformers.go [hunk 1]
```diff
@@ -180,9 +180,13 @@ func renderSidecarMountPoints(in []manifest.SidecarMountPoint) ([]*template.Moun
 	}
 	output := []*template.MountPoint{}
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/stack/transformers.go]:
```suggestion
	var output []*template.MountPoint{}
```

File: internal/pkg/deploy/cloudformation/stack/transformers.go [hunk 2]
```diff
@@ -231,9 +235,13 @@ func renderMountPoints(input map[string]manifest.Volume) ([]*template.MountPoint
...
 	for name, volume := range input {
 		// ContainerPath must be specified.
-		if volume.ContainerPath == nil {
+		if aws.StringValue(volume.ContainerPath) == "" {
 			return nil, errVolNoContainerPath
 		}
+		path := aws.StringValue(volume.ContainerPath)
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/stack/transformers.go]: `renderMountPoints` and `renderSidecarMountPoints` look pretty similar, I wonder if there is a way for us to reuse the logic?

**Review by efekarakus — APPROVED** [review]

File: internal/pkg/deploy/cloudformation/stack/workload.go [hunk 3]
```diff
@@ -34,6 +35,17 @@ const (
 	WorkloadAddonsTemplateURLParamKey = "AddonsTemplateURL"
 )
 
+// Matches alphanumeric characters and -._
```
Comment (huanjani) [inline internal/pkg/deploy/cloudformation/stack/workload.go:38]: tiny nit for the comment: and also `/`, right?

File: internal/pkg/deploy/cloudformation/stack/transformers.go (hunk 2 above)
Comment (bvtujo) [reply internal/pkg/deploy/cloudformation/stack/transformers.go]: Good idea. Addressed in my latest push.

File: internal/pkg/deploy/cloudformation/stack/workload.go [hunk 4]
```diff
@@ -224,3 +236,27 @@ func envVarOutputNames(outputs []addon.Output) []string {
 	}
 	return envVars
 }
+
+// Validate that paths contain only an approved set of characters to guard against command injection.
+// We can accept 0-9A-Za-z-_.
+func validatePath(input string, maxLength int) error {
+	if len(input) > maxLength {
+		return fmt.Errorf("path must be less than %d bytes in length", maxLength)
```
Comment (huanjani) [inline internal/pkg/deploy/cloudformation/stack/workload.go:244]: Do we want to include the path they are trying?
```suggestion
		return fmt.Errorf("path %s must be less than %d bytes in length", input, maxLength)
```

File: internal/pkg/deploy/cloudformation/stack/workload.go [hunk 5]
```diff
@@ -224,3 +236,27 @@ func envVarOutputNames(outputs []addon.Output) []string {
...
+	}
+	if len(input) == 0 {
+		return nil
+	}
+	m := pathRegexp.FindStringSubmatch(input)
+	if len(m) == 0 {
+		return fmt.Errorf("paths can only contain the characters a-zA-Z0-9.-_/")
```
Comment (huanjani) [inline internal/pkg/deploy/cloudformation/stack/workload.go:251]: Same here; maybe something like this?
```suggestion
		return fmt.Errorf("path %s can only contain the characters a-zA-Z0-9.-_/", input)
```

**Review by huanjani — APPROVED** [review]
Looks great! Two tiny nits re: error messaging. Feel free to remove label after addressing or ignoring! 😄

File: internal/pkg/deploy/cloudformation/stack/workload.go (hunk 4 above)
Comment (bvtujo) [reply internal/pkg/deploy/cloudformation/stack/workload.go:244]: For these validators I wanted to use a universal message. They're almost always wrapped by other methods, so the error responses from those calls include the path. Like so: https://github.com/aws/copilot-cli/pull/1956/files#diff-8852a201e2b451d8c315d78bf675f068153e17c99506a4dc6ecc692a25cb33cbR198

**Pull request merged**